tests/Aspire.Dashboard.Components.Tests/Controls/StateColumnDisplayTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Frozen;
+using Aspire.Dashboard.Components.ResourcesGridColumns;
+using Aspire.Dashboard.Model;
+using Google.Protobuf.WellKnownTypes;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.FluentUI.AspNetCore.Components;
+using Xunit;
+using Enum = System.Enum;
+
+namespace Aspire.Dashboard.Components.Tests.Controls;
+
+public class StateColumnDisplayTests
+{
+    private const string ResourceType = "TestResourceType";
+    private const string ExitedUnexpectedlyTooltip = "Exited Unexpectedly {0} {1}";
+    private const string ExitedTooltip = "Exited {0}";
+    private const string RunningAndUnhealthyTooltip = "Running and Unhealthy";
+    private const string UnknownStateLabel = "Unknown";
+
+    [Theory]
+    // Resource is no longer running
+    [InlineData(
+        /* state */ KnownResourceState.Exited, null, null,null,
+        /* expected output */ "Exited TestResourceType", "Warning", Color.Warning, "Exited")]
+    [InlineData(
+        /* state */ KnownResourceState.Exited, 3, null, null,
+        /* expected output */ "Exited Unexpectedly TestResourceType 3", "ErrorCircle", Color.Error, "Exited")]
+    [InlineData(
+        /* state */ KnownResourceState.Finished, 0, null, null,
+        /* expected output */ "Exited TestResourceType", "CheckmarkUnderlineCircle", Color.Success, "Finished")]
+    [InlineData(
+        /* state */ KnownResourceState.Unknown, null, null, null,
+        /* expected output */ null, "CircleHint", Color.Info, "Unknown")]
+    // Health checks
+    [InlineData(
+        /* state */ KnownResourceState.Running, null, "Healthy", null,
+        /* expected output */ null, "CheckmarkCircle", Color.Success, "Running")]
+    [InlineData(
+        /* state */ KnownResourceState.Running, null, null, null,
+        /* expected output */ null, "CheckmarkCircle", Color.Success, "Running")]
+    [InlineData(
+        /* state */ KnownResourceState.Running, null, "Unhealthy", null,
+        /* expected output */ RunningAndUnhealthyTooltip, "CheckmarkCircleWarning", Color.Neutral, "Running (Unhealthy)")]
+    [InlineData(
+        /* state */ KnownResourceState.Running, null, null, "warning",
+        /* expected output */ null, "Warning", Color.Warning, "Running")]
+    [InlineData(
+        /* state */ KnownResourceState.Running, null, null, "NOT_A_VALID_STATE_STYLE",
+        /* expected output */ null, "Circle", Color.Neutral, "Running")]
+    public void ResourceViewModel_ReturnsCorrectIconAndTooltip(
```

Comment:
@JamesNK I also added tests for these methods since there are so many different icon/tooltip/text possibilities.